### PR TITLE
Configuring Mock Responses Examples

### PR DESCRIPTION
**Pull request by gracekarina**
This PR contains:

A new feature which allows to configure the examples in the mock responses, there are two initial options:
- Random: mock a random example of the list every time the endpoint is called
- Sequence: mock a the examples sequentially from the list every time the endpoint is called.
The options are set at config file under: exampleProcessors.

This Pr also has a change, that the example response will be in the mediaType requested, as long as the media type is in the response list.

solves #212

### CODE REVIEW

File: src/main/java/io/swagger/oas/inflector/controllers/OpenAPIOperationController.java [hunk 1]
```diff
@@ -607,14 +665,28 @@ public void setContentType(RequestContext res, ResponseContext resp, Operation o
...
                     MediaType mediaType = MediaType.valueOf(key);
-                    for (MediaType acceptable : res.getAcceptableMediaTypes()) {
-                        if (mediaType.isCompatible(acceptable)) {
-                            resp.setContentType(mediaType);
-                            return;
+                    if (res.getHeaders().get("Content-Type")!= null) {
+                        for (String acceptable : res.getHeaders().get("Content-Type")) {
+                            String subtype = acceptable.substring(acceptable.lastIndexOf("/") + 1);
+
+                            if (subtype.equals(mediaType.getSubtype())) {
```
Comment (HugoMario) [inline src/main/java/io/swagger/oas/inflector/controllers/OpenAPIOperationController.java]: these statements don't make sense, the expression: `resp.setContentType(mediaType);` is executed no matter if the condition happen.

Is the conditional necessary or the `else` statement should be different?

File: src/main/java/io/swagger/oas/inflector/controllers/OpenAPIOperationController.java [hunk 2]
```diff
@@ -607,14 +665,28 @@ public void setContentType(RequestContext res, ResponseContext resp, Operation o
...
+                                resp.setContentType(mediaType);
+                            }
                         }
-                        else{
-                            resp.setContentType(mediaType);
+                    }else {
+                        for (MediaType acceptable : res.getAcceptableMediaTypes()){
+
+                            if (mediaType.isCompatible(acceptable)) {
+                                resp.setContentType(mediaType);
```
Comment (HugoMario) [inline src/main/java/io/swagger/oas/inflector/controllers/OpenAPIOperationController.java]: same here.

File: src/test/java/io/swagger/oas/test/examples/ResponseExamplesTest.java [hunk 3]
```diff
@@ -117,4 +132,193 @@ public void testResponseYamlExample(@Injectable final List<io.swagger.v3.parser.
...
+                config, "/mockResponses/objectMultipleExamples", "GET", operation, openAPI.getComponents().getSchemas() );
+
+        ContainerRequestContext requestContext = mock(ContainerRequestContext.class);
+        UriInfo uriInfo = mock( UriInfo.class );
+
+        stub( uriInfo.getPath()).toReturn( "/mockResponses/responseWithExamples");
+        stub( uriInfo.getQueryParameters()).toReturn( new MultivaluedHashMap<String, String>());
+        stub( uriInfo.getPathParameters()).toReturn( new MultivaluedHashMap<String, String>());
+
+        stub( requestContext.getAcceptableMediaTypes()).toReturn(Arrays.asList(MediaType.APPLICATION_JSON_TYPE));
```
Comment (HugoMario) [inline src/test/java/io/swagger/oas/test/examples/ResponseExamplesTest.java]: i'm wondering what happen if the content type is not any of the expected. if an error must be thrown or a response with a specific status code, can you add a test to verify the expected behave for that scenario?

File: src/main/java/io/swagger/oas/inflector/controllers/OpenAPIOperationController.java (hunk 2 above)
Comment (gracekarina) [reply src/main/java/io/swagger/oas/inflector/controllers/OpenAPIOperationController.java]: It doesn't make sense!. let me check what was I thinking back then.

File: src/test/java/io/swagger/oas/test/examples/ResponseExamplesTest.java (hunk 3 above)
Comment (gracekarina) [reply src/test/java/io/swagger/oas/test/examples/ResponseExamplesTest.java]: oki

File: src/main/java/io/swagger/oas/inflector/controllers/OpenAPIOperationController.java (hunk 2 above)
Comment (gracekarina) [reply src/main/java/io/swagger/oas/inflector/controllers/OpenAPIOperationController.java]: Done. removed the redundant code

File: src/main/java/io/swagger/oas/inflector/controllers/OpenAPIOperationController.java (hunk 1 above)
Comment (gracekarina) [reply src/main/java/io/swagger/oas/inflector/controllers/OpenAPIOperationController.java]: Done. removed the redundant code

File: src/test/java/io/swagger/oas/test/examples/ResponseExamplesTest.java (hunk 3 above)
Comment (gracekarina) [reply src/test/java/io/swagger/oas/test/examples/ResponseExamplesTest.java]: We have two test here. We changed the logic, now we won`t throw an exception, but send the response with the first mediaType of the response, (it works like that already)
here are the tests for that

https://github.com/swagger-api/swagger-inflector/blob/2.0/src/test/java/io/swagger/oas/test/integration/responses/SchemaResponseIT.java#L98-L123

**Pull request merged**